[PATCH 4/4] Fix linking with -fstack-protector
* libltdl/config/ltmain.m4sh (func_mode_link): Pass -fstack-protector*
to the linker as it is required at link time to resolve libssp symbols.
Note: this patch is necessary but not sufficient for -fstack-protector* + C++.
---
 libltdl/config/ltmain.m4sh |    4 +++-
 1 files changed, 3 insertions(+), 1 deletions(-)

diff --git a/libltdl/config/ltmain.m4sh b/libltdl/config/ltmain.m4sh
index eda7790..74ab26a 100644
--- a/libltdl/config/ltmain.m4sh
+++ b/libltdl/config/ltmain.m4sh
@@ -5086,10 +5086,12 @@ func_mode_link ()
       # -O*, -flto*, -fwhopr*, -fuse-linker-plugin GCC link-time optimization
       # -{shared,static}-libgcc, -static-{libgfortran|libstdc++}
       #                      link against specified runtime library
+      # -fstack-protector*   stack protector flags for GCC
       -64|-mips[0-9]|-r[0-9][0-9]*|-xarch=*|-xtarget=*|+DA*|+DD*|-q*|-m*| \
       -t[45]*|-txscale*|-p|-pg|--coverage|-fprofile-*|-F*|@*|-tp=*|--sysroot=*| \
       -O*|-flto*|-fwhopr*|-fuse-linker-plugin| \
-      -shared-libgcc|-static-libgcc|-static-libgfortran|-static-libstdc++)
+      -shared-libgcc|-static-libgcc|-static-libgfortran|-static-libstdc++| \
+      -fstack-protector*)
         func_quote_for_eval "$arg"
 	arg="$func_quote_for_eval_result"
         func_append compile_command " $arg"
-- 
1.7.1

